packages/vscode-extension/extension.js

```diff
@@ -30,30 +26,55 @@ class ActivefileWatcher {
 		const projectname = this.activeProject;
 		const filename = this.activefile;
 		const languageId = this.language;
+		const gitBranch = this.gitBranch;
+		const remoteurl = this.remoteUrl;
+		const projectPath = this.projectPath;
 		const data = {
 			projectname,
+			projectPath,
 			filename,
 			languageId,
+			gitBranch,
+			remoteurl,
 			startTime,
 			endTime,
 		};
 
-		console.log(data);
+		this.changeCallback(data);
 	}
 	tracker() {
 		setInterval(() => {
 			let currentProject = vscode.workspace.name;
+			let currentProjectPath = vscode.workspace.workspaceFolders[0].uri.path;
 			let currentFile = vscode.window.activeTextEditor.document.fileName;
```

Comment:
`activeTextEditor` may be undefined. Return early if it is undefined. This is a type of bug that would have been caught when using typescript.